sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -40,6 +85,17 @@ public ProxyOptions setCredentials(String username, String password) {
         return this;
     }
 
+    /**
+     * Sets the hosts which bypass the proxy.
+     *
+     * @param nonProxyHosts Hosts that bypass the proxy.
+     * @return the updated ProxyOptions object
+     */
+    public ProxyOptions setNonProxyHosts(String nonProxyHosts) {
```

Comment:
Is 'non-proxy hosts' common terminology? How does the user work with multiple non-proxy hosts? If it is, e.g. comma-separated, would it be better for the API to be varargs or a List<String>?

Reply:
`nonProxyHosts` is what Java uses for host that should ignore the environment configured proxy, I chose this to stay inline with language specifics.

Reply:
This is expected to be a regex based on the documentation for `http.nonProxyHosts` (the Java environment variable for this). Our `AZURE_NO_PROXY` environment variable doesn't state the expected format, only statement made is that is must support CIDR annotation.